packages/cli/src/runners/runner-ws-server.ts

```diff
@@ -110,4 +110,12 @@ export class TaskRunnerWsServer {
 	handleRequest(req: TaskRunnerServerInitRequest, _res: TaskRunnerServerInitResponse) {
 		this.add(req.query.id, req.ws);
 	}
+
+	async disconnect(ws: WebSocket) {
+		const runnerId = Array.from(this.runnerConnections.entries()).find(
+			([_, connection]) => connection === ws,
+		)?.[0];
+
+		if (runnerId) await this.removeConnection(runnerId);
```

Comment:
Could we provide a better error message in case this happens? Since we already know here the runner is unresponsive to pings. Now it's:

<img width="1032" alt="image" src="https://github.com/user-attachments/assets/1c2ce4fd-be78-4db2-bba6-ca08515eb096">